compiler/src/scala/quoted/runtime/impl/QuotesImpl.scala

```diff
@@ -253,6 +253,7 @@ class QuotesImpl private (using val ctx: Context) extends Quotes, QuoteUnpickler
         (cdef.name.toString, cdef.constructor, cdef.parents, cdef.self, rhs.body)
 
       def module(module: Symbol, parents: List[Tree /* Term | TypeTree */], body: List[Statement]): (ValDef, ClassDef) = {
+        xCheckMacroAssert(parents.nonEmpty && !parents.head.tpe.dealias.classSymbol.is(dotc.core.Flags.Trait), "First parent must be a class")
```

Comment:
Here we have the symbol and the parents. We could use the TreeChecker logic to check that all parents align.

Reply:
We can, but it's going to be a much stronger test. Should we do that ? Can we do that without other restrictions ?

Reply:
> Should we do that ?

It is better to have a stronger test here. This way, if the user makes a mistake they will know exactly where it happened and how to fix it. When the failure is in -Ycheck they still have to trace back to find where this issue happened.

> Can we do that without other restrictions ?

What kind of restriction?